workout-app.Application/IoC/ApplicationIoC.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using AutoMapper;
+using Autofac;
+using MediatR;
+using workout_app.Application.PipelineBehaviours;
+using MediatR.Pipeline;
+using workout_app.Application.Queries;
+using workout_app.Application.Commands;
+using FluentValidation;
+
+public class ApplicationIoC : Module
+{
+    protected override void Load(ContainerBuilder builder)
+    {
+        var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+        builder.RegisterAssemblyTypes(assemblies)
+            .Where(t => typeof(Profile).IsAssignableFrom(t) && !t.IsAbstract && t.IsPublic)
+            .As<Profile>();
+
+        builder.Register(c => new MapperConfiguration(cfg => {
+            foreach (var profile in c.Resolve<IEnumerable<Profile>>())
+            {
+                cfg.AddProfile(profile);
+            }
+        })).AsSelf().SingleInstance();
+
+        builder.Register(c => c.Resolve<MapperConfiguration>()
+            .CreateMapper(c.Resolve))
+            .As<IMapper>()
+            .InstancePerLifetimeScope();
+
+        builder.RegisterType<Mediator>().As<IMediator>().InstancePerLifetimeScope();
+
+        builder.RegisterType<GetAllExercises.GetAllExercisesHandler>().AsImplementedInterfaces().InstancePerDependency();
+        builder.RegisterType<CreateExercise.CreateExerciseHandler>().AsImplementedInterfaces().InstancePerDependency();
+        builder.RegisterType<GetExerciseById.GetExerciseByIdHandler>().AsImplementedInterfaces().InstancePerDependency();
```

Comment:
Żeby uniknac rejestrowania wszystkich handlerów po kolei możesz użyc refleksji

```suggestion
builder.RegisterAssemblyTypes(typeof(ApplicationIoC).GetTypeInfo().Assembly).AsClosedTypesOf(typeof(IRequestHandler<,>));
```

Reply:
Dobry pomysł ;) zakomitowałem zmiany